tapgarden/interface.go

```diff
@@ -126,7 +126,39 @@ func (b BatchState) String() string {
 		return "BatchStateSproutCancelled"
 
 	default:
-		return fmt.Sprintf("UnknownState(%v)", int(b))
+		return fmt.Sprintf("UnknownState(%v)", uint32(b))
+	}
+}
+
+func NewBatchState(state uint32) (BatchState, error) {
+	switch state {
```

Comment:
I think you can just do `switch BatchState(state) {` to avoid all the `uint32` casts below.